packages/medusa/src/strategies/batch-jobs/product/import.ts

```diff
@@ -463,6 +486,53 @@ class ProductImportStrategy extends AbstractBatchJobStrategy {
     }
   }
 
+  private async assignProductToCollection(
+    product: Product,
+    collectionData: { title: string; handle: string }
+  ): Promise<void> {
+    const transactionManager = this.transactionManager_ ?? this.manager_
+    const productCollectionServiceTx =
+      this.productCollectionService_.withTransaction(transactionManager)
+
+    if (product.collection_id) {
+      const currentCollection = await productCollectionServiceTx.retrieve(
+        product.collection_id,
+        {
+          select: ["handle"],
+        }
+      )
+
+      if (currentCollection.handle === collectionData.handle) {
+        return
+      }
+
+      await productCollectionServiceTx.removeProducts(product.collection_id, [
+        product.id,
+      ])
+    }
+
+    let shouldCreateCollection = false
+
+    let collection = await productCollectionServiceTx
+      .retrieveByHandle(collectionData.handle, { select: ["id"] })
+      .catch((e) => {
+        if ("type" in e && e.type === MedusaError.Types.NOT_FOUND) {
+          shouldCreateCollection = true
+          return
+        }
+        throw e
+      })
+
+    if (shouldCreateCollection) {
```

Comment:
context: previous discussion about this - https://linear.app/medusajs/issue/CORE-700#comment-59937c68

Reply:
Alright, any reason for that? The reason is not mentioned in the ticket

Reply:
I have removed that part 🎉